"make install" doesn't work anymore

the archlinux package is broken, and I'm not sure what's going on exactly. After adding `make lib-ext`, it builds (more or less) but i can't install, because of : 

```
src/opam-installer --prefix /usr opam.install
src/opam                         => /usr/bin/opam
Fatal error: exception OpamSystem.Process_error(_)
Makefile:34: recipe for target 'install' failed
make: *** [install] Error 2
```

cc @vbmithr

--- a/Makefile
+++ b/Makefile
@@ -25,19 +25,29 @@ download-ext:
 clean-ext:
 	$(MAKE) -C src_ext distclean
 
-LIBINSTALL_PREFIX = $(dir $(shell $(OCAMLFIND) printconf destdir))
+ifdef DESTDIR
+  LIBINSTALL_PREFIX ?= $(DESTDIR)
+else
+  DESTDIR ?= $(prefix)
+  ifneq ($(OCAMLFIND),no)
+    LIBINSTALL_PREFIX = $(dir $(shell $(OCAMLFIND) printconf destdir))
+  else
+    LIBINSTALL_PREFIX ?= $(DESTDIR)
+  endif
+endif
+
 libinstall:
 	$(if $(wildcard src_ext/lib/*),$(error Installing the opam libraries is incompatible with embedding the dependencies. Run 'make clean-ext' and try again))
 	src/opam-installer --prefix $(LIBINSTALL_PREFIX) --name opam opam-lib.install
 
 install:
-	src/opam-installer --prefix $(prefix) opam.install
+	src/opam-installer --prefix $(DESTDIR) opam.install
 
 libuninstall:
 	src/opam-installer -u --prefix $(LIBINSTALL_PREFIX) --name opam opam-lib.install
 
 uninstall:
-	src/opam-installer -u --prefix $(prefix) opam.install
+	src/opam-installer -u --prefix $(DESTDIR) opam.install
 
 .PHONY: tests tests-local tests-git
 tests:

